Functions/Example/IntegrationTests/FIRIntegrationTests.m

```diff
@@ -77,7 +81,10 @@ - (void)testScalar {
 - (void)testToken {
   // Recreate _functions with a token.
   id app = [[FUNFakeApp alloc] initWithProjectID:@"functions-integration-test" token:@"token"];
-  FIRFunctions *functions = [FIRFunctions functionsForApp:app];
+  FIRFunctions *functions = [[FIRFunctions alloc]
+      initWithApp:app
+           region:@"my-region"
+             auth:[[FIRAuthInteropFake alloc] initWithToken:nil userID:nil error:nil]];
```

Comment:
I don't understand why this passes. Shouldn't functions be using the `FIRAuthInteropFake`'s token instead of the one passed into the `FUNFakeApp`?

Reply:
I'm also unsure how this was passing.

It looks like [`callFunction`](https://github.com/firebase/firebase-ios-sdk/blob/master/Functions/FirebaseFunctions/FIRFunctions.m#L163) would have an empty Authorization header, but maybe the mock server still returns an empty data response with a 200 OK?

Reply:
Actually, I bet it wasn't passing. Travis doesn't run the integration tests on PRs, just the master branch. I'll run them locally to make sure they're passing now.

Reply:
I'm having difficulty running the integration tests. I get many warnings when running [`Backend/start.sh` instructions](https://github.com/firebase/firebase-ios-sdk/tree/master/Functions). See this [gist](https://gist.github.com/bstpierr/9c69a40e7c5f55142671f6e9575c1bf5) of the output I get.

Reply:
I changed my system to node 8, and ran through the suggestions it printed out, and it seems as though it's connecting to the emulator, but I'm not sure it's behaving as expected.

For example, the `testData` test case gets the error: `Function dataTest in location my-region in project functions-integration-test does not exist` in its [`FIRHTTPSCallableResult` callback](https://github.com/firebase/firebase-ios-sdk/blob/2c9f881b674ebf9b516f63c0b7ec78d9dd61fabc/Functions/Example/IntegrationTests/FIRIntegrationTests.m#L56).

From the emulator's output, it looks like `my-region` isn't what's expected, but rather `us-central1` since it outputs `http://localhost:5005/functions-integration-test/us-central1/dataTest` as the resource.

Switching the tests to using `us-central1` as the region has all the tests but one passing. The [`testNull`](https://github.com/firebase/firebase-ios-sdk/blob/2c9f881b674ebf9b516f63c0b7ec78d9dd61fabc/Functions/Example/IntegrationTests/FIRIntegrationTests.m#L106) attempts to wait on an [expectation twice](https://github.com/firebase/firebase-ios-sdk/blob/2c9f881b674ebf9b516f63c0b7ec78d9dd61fabc/Functions/Example/IntegrationTests/FIRIntegrationTests.m#L124), which is an API violation that throws an error.

Are the integration tests expected to pass? I'm not sure how they would on master without the region change and the duplicate `waitForExpectation:` call.

Reply:
It appears as though the [last green run of master](https://travis-ci.org/firebase/firebase-ios-sdk/jobs/464154830) compiled the integration tests, but only ran the unit tests.

It seems as though the integration tests [aren't being run on Travis](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/build.sh#L191).

I'll open an internal bug to fix and enable theses.

Reply:
> From the emulator's output, it looks like my-region isn't what's expected, but rather us-central1

Yes, the shell script in Backend/ that installs the functions says to install them in `us-central1`. Not sure when that broke.